### PR TITLE
Lab8

### PR DESCRIPTION
**Pull request by Andrew174194**
Pull request for lab 8
by Andrey Khoroshavin (kill team, CS-01 academic group)

### CODE REVIEW

File: monitoring/docker-compose.yml [hunk 1]
```diff
@@ -3,38 +3,52 @@ version: "3"
 x-logging:
   &logs
   options:
+    max-size: "8m"
+    max-file: "4"
     tag: "{{.Name}}"
 
+x-healthcheck:
+  &healthcheck
+  test: echo "No test function provided" && exit 1 # is required to be overrided
```
Comment (DOCtorActoAntohich) [inline monitoring/docker-compose.yml:12]: kinda based :trollface:

File: monitoring/docker-compose.yml [hunk 2]
```diff
@@ -46,11 +60,33 @@ services:
       - grafana
       - python_app
     logging: *logs
+    healthcheck:
+      << : *healthcheck
+      test: timeout 1 bash -c 'cat < /dev/null > /dev/tcp/localhost/9080' || exit 1
 
   grafana:
     image: grafana/grafana:latest
+    mem_limit: 128m
```
Comment (DOCtorActoAntohich) [inline monitoring/docker-compose.yml:69]: MEME LIMIT :lets:

**Review by DOCtorActoAntohich — APPROVED** [review]
lgtm (legendary gregification type mongus)

File: app_python/main.py [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
 
 # for metrics
-startTime = time.time()
-responses = 0
+from prometheus_client import Counter, Histogram, generate_latest
+CONTENT_TYPE = str('text/plain; version=0.0.4; charset=utf-8')
+REQUEST_COUNT = Counter(
+    'request_count', 'App Request Count',
+    ['app_name', 'method', 'endpoint', 'http_status']
+)
```
Comment (ntdesmond) [inline app_python/main.py:12]: lol manual metric collection good

File: monitoring/docker-compose.yml (hunk 1 above)
Comment (ntdesmond) [reply monitoring/docker-compose.yml:12]: cool btw

**Review by ntdesmond — APPROVED** [review]
e

**Pull request closed without merging**